### PR TITLE
Moved writable validation to write methods in FileCache impl

### PR DESCRIPTION
**Pull request by brzuchal**
Those changes are adoption to non-writable environments (eg. in Docker container images) and all writability checks are moved into methods which writes data - this is needed when we do not have permissions to write, this situation is covered by tests using vfsStream - avoid filesystem access while contributing.

There was also moved `rename` and `chmod` from `renameFile` method into `putClassMetadataInCache` because it wasn't used elsewere.

May sound obvious but added also `phpunit` as dev dependency.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
-vendor
\ No newline at end of file
+vendor
+/.idea/
```
Comment (staabm) [inline .gitignore]: IDE/Editor related ignores should be handled globally in your git config, not per project

Comment (dgafka) [inline .gitignore]: Done.

**Comment by dgafka** [comment]
Please take a look at the MR.

**Comment by jurgenf** [comment]
As I am preparing to move my projects to read-only containers I would be very much helped if the proposed change the move the write-ability checks would be merged into the next release.

**Comment by jaapio** [comment]
ping @schmittjoh. It would be very nice if this PR could be merged. It looks quite good to me. And it solves an issue we have right now in phpdocumentor with the phar release.

**Comment by goetas** [comment]
this might help https://github.com/schmittjoh/JMSSerializerBundle/pull/615

**Comment by jaapio** [comment]
It does but, because the cache dir is in a phar file which is readonly and the constructor checks if this dir is writable. Which is not.

**Comment by goetas** [comment]
ah, i see. This PR is pretty old and with conflits. If you want you can rebase it on the `1.x` branch and we can give it a second chance.

**Comment by bobvandevijver** [comment]
Just to revive this MR (@goetas @schmittjoh @brzuchal), this change would most probably solve the issue the serializerbundle (https://github.com/schmittjoh/JMSSerializerBundle/issues/823) and vich uploader (https://github.com/dustin10/VichUploaderBundle/issues/1161) have with Symfony 5.2 (https://github.com/symfony/symfony/issues/39232).

Can we update, merge and release this?

**Comment by goetas** [comment]
@bobvandevijver  if you are interested into solve this issue, can you please rebase this on top of the latest branch? the work in this PR was done 4 years ago and many other changes landed on the latest branch (after that I can  merge it if there are no issues..)

**Comment by brzuchal** [comment]
Closing since there is a fresh PR pending.
I no longer use this package and unfortunately can't spend time on it in near future.

**Pull request closed without merging**